core/src/main/scala/io/finch/ToService.scala

```diff
@@ -94,19 +97,25 @@ object ToService {
     def apply(
         endpoints: Endpoint[A] :: ET,
         includeDateHeader: Boolean,
-        includeServerHeader: Boolean): Service[Request, Response] = new Service[Request, Response] {
+        includeServerHeader: Boolean,
+        methodNotAllowed: Boolean): Service[Request, Response] = new Service[Request, Response] {
 
       private[this] val underlying = endpoints.head.handle(respond400OnErrors)
 
       def apply(req: Request): Future[Response] = underlying(Input.fromRequest(req)) match {
-        case EndpointResult.Matched(rem, out) if rem.route.isEmpty => out.map(oa => conformHttp(
-          oa.toResponse(trA, trE),
-          req.version,
-          includeDateHeader,
-          includeServerHeader
-        )).run
-
-        case _ => tsT(endpoints.tail, includeDateHeader, includeServerHeader)(req)
+        case EndpointResult.Matched(rem, out) if rem.route.isEmpty =>
```

Comment:
How about we move that method check under the if-guard? Something along the lines:

```scala
case EndpointResult.Matched(rem, out) if (rem.route.isEmpty && rem.method == req.method) =>
  // full match
case EndpointResult.Matched(rem, out) if (rem.route.isEmpty) =>
  // method wasn't matched
case _ => // fall through
```

Reply:
Then it will have to unapply it twice, so I think it's not that efficient compared to current solution.

Reply:
It will be fine - unapply is supper for case classes (it just a couple of `isInstanceOf`/ `asInstanceOf` calls).